### DISCUSSION TITLE
Using Arduino_GFX and sprites or canvas.

### DISCUSSION THREAD

**Discussion by JeffreyRon** [moononournation/Arduino_GFX#369] [Q&A]
Im just getting up to speed with Arduino_GFX and I want to do sprites. Or perhaps canvas. I'm having trouble understanding what to use with Arduino_GFX. I know more about TFT_eSPI and sprites. Anybody got some hello world type sprites type functionality for Arduino_GFX.

**Comment by moononournation** [comment]
it have a built-in example call spritegif

**Reply by JeffreyRon** [reply]
Thanks for your response. Unfortunately that example is way over my head with the indexing etc. I need to start simple.  I've been trying various things and can't seem to get the actual displaying of the canvas syntax correct. The TFT_eSPI sprites are so simple. Maybe I'm just missing something simple here?

What I've tried...

(this compiles..........)

Arduino_RGB_Display *gfx = new Arduino_RGB_Display(800 /* width */, 480 /* height */, rgbpanel, 0 /* rotation */, true /* auto_flush */);
Arduino_GFX *canvasGfx = new Arduino_Canvas(800 /* width */, 480 /* height */, gfx);                                                        // 16bit

  gfx->begin();
  canvasGfx->begin();

  gfx->setTextColor(WHITE);
  gfx->drawRect(0, 0, 800, 480, ORANGE);
  gfx->println("Hello World!1");

  canvasGfx->setTextColor(RED, BLUE);
  canvasGfx->fillScreen(BLACK);
  canvasGfx->drawCircle(400, 240, 60, RED);
  canvasGfx->println("Hello World2");

But then when it's time to actually display the canvas, I've tried all of these and many more and none compile.... 

  //gfx->draw(canvasGfx);
  gfx->drawRGBBitmap(0, 0, canvasGfx->getBuffer(), 800, 480)
  //gfx->draw3bitRGBBitmap(0, 0, canvasGfx->getBuffer(), 800, 480)  //when using the 3bit initialization
  //gfx->draw3bitRGBBitmap(0, 0, canvasGfx.getBuffer(), 800, 480)  //when using the 3bit initialization
  //gfx->drawBitmap(0, 0, canvasGfx, 800, 480)

**Reply by moononournation** [reply]
if you want to use Arduino_GFX with sprite, spritegif example is the beginning. You may not know how it works, but simply try to run the example first.
